cartridge/lua-api/get-topology.lua

```diff
@@ -65,8 +68,15 @@ local confapplier = require('cartridge.confapplier')
 -- @treturn[2] table Error description
 local function get_topology()
     local state, err = confapplier.get_state()
-    -- OperationError doesn't influence observing topology
-    if state == 'InitError' or state == 'BootError' then
+    if state == 'Unconfigured' and lua_api_proxy.can_call() then
+        -- Try to proxy call
+        local ret = lua_api_proxy.call(mod_name .. '.get_topology')
+        if ret ~= nil then
+            return ret
+        -- else
+            -- Don't return an error, go on
+        end
+    elseif state == 'InitError' or state == 'BootError' then
```

Comment:
I see similar code at least two times. Maybe it's reasonable to move it to separate function?